src/drivers/distance_sensor/ll40ls/LidarLiteI2C.cpp

```diff
@@ -128,35 +128,56 @@ LidarLiteI2C::probe()
 
 		set_device_address(addresses[i]);
 
-		if (addresses[i] == LL40LS_BASEADDR) {
+		/*
```

Comment:
```suggestion
		/*
		 * The probing is divided in two cases. One to detect v1, v2 and v3 and one to
		 * detect v3HP. The reason for this is that registers are not consistent
		 * between different versions. The v3HP doesn't have the HW VERSION (or at least no version is specified)
		 * while v1, v2 and some v3 don't have the unit id register.
		 * It would be better if we had a proper WHOAMI register.
		 */
```